engine/launcher/src/main/scala/org/enso/launcher/PluginManager.scala

```diff
@@ -0,0 +1,151 @@
+package org.enso.launcher
+
+import java.io.File
+import java.nio.file.{Files, Path}
+
+import sys.process._
+import org.enso.cli.{
+  CommandHelp,
+  PluginBehaviour,
+  PluginInterceptedFlow,
+  PluginManager,
+  PluginNotFound
+}
+
+import scala.collection.Factory
+import scala.util.Try
+import scala.jdk.StreamConverters._
+import scala.util.control.NonFatal
+
+/**
+  * An abstraction of the system environment.
+  */
+trait SystemEnvironment {
+
+  /**
+    * Returns the system PATH, if available.
+    */
+  def getSystemPath: Option[String]
+
+  /**
+    * Returns a list of system-dependent plugin extensions.
+    *
+    * By default, on Unix plugins should have no extensions. On Windows, `.exe`
+    * and `.bat` are supported.
+    */
+  def getPluginExtensions: Seq[String] =
+    if (System.getProperty("os.name").toLowerCase.contains("windows"))
+      Seq("bat", "exe")
+    else Seq()
+
+}
+
+/**
+  * Implements a [[PluginManager]] using the provided [[SystemEnvironment]].
+  */
+class PluginManagerImplementation(environment: SystemEnvironment)
+    extends PluginManager {
+
+  /**
+    * Checks if the provided name represents a valid plugin and tries to run it.
+    * @param name name of the plugin
+    * @param args arguments that should be passed to it
+    * @return
+    */
+  override def tryRunningPlugin(
+    name: String,
+    args: Seq[String]
+  ): PluginBehaviour =
+    if (hasPlugin(name)) {
+      try {
+        val exitCode = (Seq(pluginCommandForName(name)) ++ args).!
+        System.exit(exitCode)
+        PluginInterceptedFlow
+      } catch {
+        case NonFatal(_) =>
+          System.err.println(
```

Comment:
Is this something we want to show users? We probably want to think about logging it too.

Reply:
I thought that it may indicate some unexpected configuration issue (now that I think about it I'm not even sure if it is possible that an exception is thrown here) and thought that it may be better to inform the user (this can only happen if the plugin is successfully found on the path but fails to execute).

Reply:
Yeah that's fair enough! Keep it.